packages/vite/src/node/server/transformRequest.ts

```diff
@@ -249,6 +250,11 @@ async function loadAndTransform(
     isDebug && debugTransform(`${timeFrom(transformStart)} ${prettyUrl}`)
     code = transformResult.code!
     map = transformResult.map
+
+    // To enable debugging create a sourcemap for known modified JS files without one:
+    if (!map && mod.file && mod.type === 'js' && code !== originalCode) {
+      map = new MagicString(code).generateMap({ source: mod.file })
+    }
   }
 
```

Comment:
I think we need to change line 262 to support browser debugger:
```js
    if (
      map.mappings &&
      (!map.sourcesContent ||
        (map.sourcesContent as Array<string | null>).includes(null))
    ) {
```

Reply:
Right, makes sense. Thanks! Added the change.